### PR TITLE
Add Support for Admins to Contribute to Past Events

### PR DESCRIPTION
**Pull request by SudharakaP**
Resolve https://github.com/opencollective/opencollective/issues/3982

Requires: https://github.com/opencollective/opencollective-api/pull/6511 

Related to https://github.com/opencollective/opencollective-frontend/pull/6795

https://user-images.githubusercontent.com/12435965/131178317-71fa43f5-f50a-41ed-ab1a-6b50f2f5bc2e.mp4

### CODE REVIEW

File: lib/collective-sections.js [hunk 1]
```diff
@@ -160,7 +160,7 @@ const showContributeSection = (collective, isAdmin) => {
     case 'AVAILABLE':
       return true;
     case 'DISABLED':
-      return collective.type === CollectiveType.COLLECTIVE && isAdmin;
+      return (collective.type === CollectiveType.COLLECTIVE || collective.type === CollectiveType.EVENT) && isAdmin;
```
Comment (znarf) [inline lib/collective-sections.js:163]: `FUND` and `PROJECT` also.

Maybe even all `type`s in the future, but not for now.

File: components/collective-page/sections/Contribute.js [hunk 2]
```diff
@@ -161,9 +161,10 @@ class SectionContribute extends React.PureComponent {
   });
 
   getFinancialContributions = memoizeOne(sortedTiers => {
-    const { collective, contributors, contributorsStats } = this.props;
+    const { collective, contributors, contributorsStats, isAdmin } = this.props;
     const hasNoContributor = !this.hasContributors(contributors);
-    const canContribute = collective.isActive && !isPastEvent(collective);
+    const canContribute =
+      (collective.isActive && !isPastEvent(collective)) || (collective.type === CollectiveType.EVENT && isAdmin);
```
Comment (znarf) [inline components/collective-page/sections/Contribute.js:167]:
```suggestion
      (collective.isActive && (!isPastEvent(collective) || isAdmin));
```

**Comment by SudharakaP** [comment]
@znarf : Thanks for the reviews. I've done the changes based on those. 👍🏽

**Comment by SudharakaP** [comment]
@znarf : Just to note, as per Ben's suggestion here; https://opencollective.slack.com/archives/C020ZQSLM5M/p1630321759013100?thread_ts=1630091991.008600&cid=C020ZQSLM5M I've added a small warning to show for past events. Here's a screencast which show the contribution flow for a past event and a regular event; 

past event warning

File: lib/collective-sections.js [hunk 3]
```diff
@@ -160,7 +160,13 @@ const showContributeSection = (collective, isAdmin) => {
     case 'AVAILABLE':
       return true;
     case 'DISABLED':
-      return collective.type === CollectiveType.COLLECTIVE && isAdmin;
+      return (
+        (collective.type === CollectiveType.COLLECTIVE ||
+          collective.type === CollectiveType.EVENT ||
+          collective.type === CollectiveType.FUND ||
+          collective.type === CollectiveType.PROJECT) &&
```
Comment (znarf) [inline lib/collective-sections.js:167]: Please use a more elegant syntax, use `includes`.

File: components/contribution-flow/index.js [hunk 4]
```diff
@@ -87,13 +89,18 @@ const OTHER_MESSAGES = defineMessages({
     defaultMessage:
       'You are about to make a contribution of {contributionAmount} to {accountName}, with a tip to the Open Collective platform of {tipAmount}. This means the tip is larger than the contribution, when usually the reverse is intended.{newLine}{newLine}Are you sure you want to do this?',
   },
+  pastEventWarning: {
+    id: 'Warning.PastEvent',
+    defaultMessage: `You're contributing to a past event`,
```
Comment (znarf) [inline components/contribution-flow/index.js:94]:
```suggestion
    defaultMessage: `You're contributing to a past event.`,
```

**Review by znarf — CHANGES_REQUESTED** [review]

File: lib/collective-sections.js (hunk 3 above)
Comment (SudharakaP) [reply lib/collective-sections.js]: Done. 👍🏽

**Review by znarf — APPROVED** [review]
Good for me. Next step would be to get team feedback and see if we can remove this restriction for non-admins too. But make sure to disable ticket buying for sure.

**Comment by SudharakaP** [comment]
> Good for me. Next step would be to get team feedback and see if we can remove this restriction for non-admins too. But make sure to disable ticket buying for sure.

@znarf : Thanks for the feedback. I'll be merging this today then. As for allowing it for non-admins (without ticket buying), I'll create another issue and then we'll discuss in tomorrow standup. 👍🏽

**Comment by SudharakaP** [comment]
> @znarf : Thanks for the feedback. I'll be merging this today then. As for allowing it for non-admins (without ticket buying), I'll create another issue and then we'll discuss in tomorrow standup. 👍🏽

And the issue is created at, https://github.com/opencollective/opencollective/issues/4617

**Pull request merged**